### PR TITLE
Update to PHP 7.4 + cleanup

### PR DESCRIPTION
**Pull request by webrouse**
Changes:
- Updated from PHP `7.3` -> `7.4`
- Code cleanup
- Updated Dockerfile to match php-component template
- Fixed types in code

### CODE REVIEW

File: src/Keboola/DbExtractor/Extractor/MSSQL.php [hunk 1]
```diff
@@ -451,7 +451,7 @@ function (array $column): string {
             $this->db->quoteIdentifier($table['tableName'])
         );
 
-        if ($table['nolock']) {
+        if ($table['nolock'] ?? false) {
```
Comment (webrouse) [inline src/Keboola/DbExtractor/Extractor/MSSQL.php:454]: Kluc nemusi byt definovany, kedze tam nebolo `php-prod.ini` a zapnuty `E_STRICT`, tak to predtym nepadalo.

File: tests/Keboola/DbExtractor/AbstractMSSQLTest.php [hunk 2]
```diff
@@ -13,15 +14,15 @@ abstract class AbstractMSSQLTest extends ExtractorTest
 {
     public const DRIVER = 'mssql';
 
-    /** @var \PDO */
+    /** @var PDO */
     protected $pdo;
 
-    /** @var string  */
+    /** @var string */
     protected $dataDir = __DIR__ . '/../../data';
```
Comment (ondrajodas) [inline tests/Keboola/DbExtractor/AbstractMSSQLTest.php:21]: proč ne `protected PDO $pdo` a `protected string $datadir`?

Comment (webrouse) [reply tests/Keboola/DbExtractor/AbstractMSSQLTest.php:21]: @ondrajodas `$datadir` musi byt taky ako v nadradenej triede, ...  a tam je (zatial) stara verzia `db-extractor-common` .... 

`PDO` je vzdy definovane .... no v `setUp` je podmienka:
https://github.com/keboola/db-extractor-mssql/blob/13aeeea53bd0d54f416ecfc8d4330b46476c3410/tests/Keboola/DbExtractor/AbstractMSSQLTest.php#L25

A teda tam nemoze byt definovane `protected PDO $pdo;` lebo to spadne v tej podmienke (`PDO must be defined before ...`) .... ak by som dal `protected ?PDO $pdo;`, tak zase v celych testoch vysoci chyba ze volam metody nad `null|PDO` .... teda bez nejakej vacsej upravy testov (co som v tomto PR nechcel robit) sa to inak napisat neda.

Comment (ondrajodas) [reply tests/Keboola/DbExtractor/AbstractMSSQLTest.php:21]: tak to potom OK :)

**Pull request merged**